### PR TITLE
Document ConsoleControl

### PR DESCRIPTION
**Pull request by zadjii-msft**
This is fairly rudimentary documentation for a function that I think few of us truly understand anymore. 

* [ ] TODO: I don't know what `ms.assetid` should be

### CODE REVIEW

File: docs/consolecontrol.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+---
+title: ConsoleControl function
```
Comment (DHowett) [inline docs/consolecontrol.md:2]: You might need to add it to the TOC as well (toc.yml)

File: docs/consolecontrol.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+- ConsoleControl
+api_location:
+- User32.dll
+api_type:
+- DllExport
+---
+
+# ConsoleControl function
+
+Performs special kernel operations for console host applications. This includes: reparenting the console window, allowing the console to give launched applications foreground rights, and terminating attached processes.
```
Comment (DHowett) [inline docs/consolecontrol.md:33]:
```suggestion
Performs special kernel operations for console host applications. This includes reparenting the console window, allowing the console to pass foreground rights on to launched console subsystem applications and terminating attached processes. 
```

reword 2, kill oxford comma 3

File: docs/consolecontrol.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+- User32.dll
+api_type:
+- DllExport
+---
+
+# ConsoleControl function
+
+Performs special kernel operations for console host applications. This includes: reparenting the console window, allowing the console to give launched applications foreground rights, and terminating attached processes. 
+
+<div class="alert"><b>Note</b> This function has no associated import library. This function is available as the resources named <b>ConsoleControl</b> in User32.dll. You must use the <a href="/windows/desktop/api/libloaderapi/nf-libloaderapi-loadlibrarya">LoadLibrary</a> and <a href="/windows/desktop/api/libloaderapi/nf-libloaderapi-getprocaddress">GetProcAddress</a> functions to dynamically link to User32.dll.</div><div> </div>
```
Comment (DHowett) [inline docs/consolecontrol.md:35]:
```suggestion
<div class="alert"><b>Note</b> This function has no associated import library. This function is available as the resource named <b>ConsoleControl</b> in User32.dll. You must use the <a href="/windows/desktop/api/libloaderapi/nf-libloaderapi-loadlibrarya">LoadLibrary</a> and <a href="/windows/desktop/api/libloaderapi/nf-libloaderapi-getprocaddress">GetProcAddress</a> functions to dynamically link to User32.dll.</div><div> </div>
```

File: docs/consolecontrol.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+## Remarks
+
+This function is not defined in an SDK header and must be declared by the caller. This function is exported from user32.dll
+
+## Requirements
+
+| &nbsp; | &nbsp; |
+|-|-|
+| Minimum supported client | Windows 7 \[desktop apps only\] |
+| Minimum supported server | Windows Server 2003 \[desktop apps only\] |
```
Comment (DHowett) [inline docs/consolecontrol.md:108]: are these right?

File: docs/consolecontrol.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+<div class="alert"><b>Note</b> This function has no associated import library. This function is available as the resources named <b>ConsoleControl</b> in User32.dll. You must use the <a href="/windows/desktop/api/libloaderapi/nf-libloaderapi-loadlibrarya">LoadLibrary</a> and <a href="/windows/desktop/api/libloaderapi/nf-libloaderapi-getprocaddress">GetProcAddress</a> functions to dynamically link to User32.dll.</div><div> </div>
+
+## Syntax
+
+```C
+
+typedef enum _CONSOLECONTROL
+{
+    Reserved1,
+    ConsoleNotifyConsoleApplication,
```
Comment (DHowett) [inline docs/consolecontrol.md:44]: Are there structs for Notify and Caret?

File: docs/consolecontrol.md (hunk 2 above)
Comment (DHowett) [reply docs/consolecontrol.md:33]: (I reworded 2 because it sounds too much like "the console can break all of user32's foreground rules")

Comment (DHowett) [reply docs/consolecontrol.md:33]: Maybe "kernel operations" -> "Windows subsystem operations" (not all of win32 is in the kernel)

File: docs/consolecontrol.md (hunk 4 above)
Comment (zadjii-msft) [reply docs/consolecontrol.md:108]: Literally no idea.

Comment (zadjii-msft) [reply docs/consolecontrol.md:108]: Actually, yes these are right. They're in the Win7 codebase, not Vista. That lines up with the console/csrss split (if my memory serves me right)

**Review by DHowett — APPROVED** [review]

File: docs/consolecontrol.md (hunk 4 above)
Comment (DHowett) [reply docs/consolecontrol.md:127]: server 2003 is XP, FYI

File: docs/consolecontrol.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+* `ConsoleSetForeground`: expects a pointer to a `CONSOLESETFOREGROUND`
+* `ConsoleSetWindowOwner`: expects a pointer to a `CONSOLEWINDOWOWNER`
+* `ConsoleEndTask`: expects a pointer to a `CONSOLEENDTASK`
+
+## Requirements
+
+| &nbsp; | &nbsp; |
+|-|-|
+| Minimum supported client | Windows 7 \[desktop apps only\] |
+| Minimum supported server | Windows Server 2003 \[desktop apps only\] |
```
Comment (DHowett) [inline docs/consolecontrol.md:127]:
```suggestion
| Minimum supported server | Windows Server 2008 R2 \[desktop apps only\] |
```

**Review by DHowett — CHANGES_REQUESTED** [review]
noted inline

**Review by DHowett — APPROVED** [review]

**Pull request merged**